### PR TITLE
chore(ESSNTL-4076): upgrade react-router to V6, refactor all class c…

### PR DESCRIPTION
**Pull request by mkholjuraev**
…omponents

Resolves: https://issues.redhat.com/browse/ESSNTL-4076

Upgrades react-router to v6 and refactors all class components to function components

### CODE REVIEW

**Comment by Fewwy** [comment]
The inventory table works, but when I'm trying to open it in the RHEL advisor - it fails to load. Is this expected?

**Comment by mkholjuraev** [comment]
@Fewwy are you running both apps together?

**Comment by mkholjuraev** [comment]
@Fewwy The tests are passing, can I ask for your re-review?

**Review by Fewwy — APPROVED** [review]

**Comment by mkholjuraev** [comment]
@Fewwy thank you for the review. I am doing the last check on how to sync the router upgrades between inventory and other dependent apps. Let's hold on with merging this for now.

File: src/App.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
             event => {
                 if (event.navId === 'inventory') {
-                    history.push(`/${location.search}${location.hash}`);
+                    navigate(`/${location.search}${location.hash}`);
                 } else {
-                    history.push(`/${event.navId}${location.search}${location.hash}`);
+                    navigate(`/${event.navId}${location.search}${location.hash}`);
                 }
             }
         );
```
Comment (bastilian) [inline src/App.js:22]: We can and should remove this according to the platform team.

**Comment by mkholjuraev** [comment]
Closing as it takes more effort than starting from scratch

**Pull request closed without merging**